Makefile

```diff
@@ -38,12 +38,30 @@ lint: lint.sub
 push: push.sub
 test: test.sub
 
+clean:
```

Comment:
That should work. To be honest, when I tried to figure out how "make clean" worked I got lost in the recursive nature of things. Best I could figure out was that the nested Makefiles would delete the generated binaries - but I couldn't find a spot to add a custom "rm" type of command - so I create this.

I could be wrong, but I believe 'make clean' will do both - the nested "clean" and this dedicated one.  But I like your solution because then its clear that both will happen w/o any make magic.

Fixed.